apps/site/assets/ts/stop/components/DepartureTimes.tsx

```diff
@@ -291,19 +173,22 @@ const DepartureTimes = ({
     directionId
   );
 
-  const schedules = schedulesByHeadsign(schedulesForDirection);
+  // [headsign, Schedules[]]
+  const schedulesForHeadsign = schedulesByHeadsign(schedulesForDirection);
 
   return (
     <>
-      {Object.entries(schedules).map(([headsign, schs]) => {
-        const formattedTimes = toDisplayTime(
+      {Object.entries(schedulesForHeadsign).map(([headsign, schs]) => {
+        const predictions = predictionsByHeadsign[headsign]
+          ? predictionsByHeadsign[headsign]
+          : [];
```

Comment:
nitpick (non-blocking): Could be simplified to `predictionsByHeadsign[headsign] || []`